frontend/class-opengraph.php

```diff
@@ -681,10 +681,9 @@ public function category() {
 
 		$terms = get_the_category();
 
-		if ( ! is_wp_error( $terms ) && ( is_array( $terms ) && $terms !== array() ) ) {
+		if ( ! is_wp_error( $terms ) && is_array( $terms ) && ! empty( $terms ) ) {
 			// We can only show one section here, so we take the first one.
-			$this->og_tag( 'article:section', $terms[0]->name );
-
+			$this->og_tag( 'article:section', reset( $terms )->name );
```

Comment:
Hi @andizer - Sure, there are several ways to skin this cat, I just picked the most concise version, and intuitively chose `reset()` over `array_shift()` because `reset()`just changes the  internal _"current"_ pointer and `$term` actually modifies the `$terms` array which is a memory and time heavier approach.

I was curious so I benchmarked both and found `reset()` to be `19%` faster than `array_shift()`, but doing it once the difference is so infinitesimal that it does not really matter which way it is done.

Same with copying to a variable. It is were me for shipping code where most people are users vs. readers of the code I would stick with `reset()` but since its your repo I'll make the requested update.

Reply:
Done.